app/src/main/java/eu/kanade/tachiyomi/ui/reader/viewer/pager/PagerPageHolder.kt

```diff
@@ -155,58 +157,74 @@ class PagerPageHolder(
      * Called when the page is ready.
      */
     private suspend fun setImage() {
-        if (extraPage == null) {
-            progressIndicator.setProgress(0)
-        } else {
-            progressIndicator.setProgress(95)
-        }
-        errorLayout?.root?.isVisible = false
-
-        val streamFn = page.stream ?: return
-        val streamFn2 = extraPage?.stream
-
-        val (bais, isAnimated, background) = withIOContext {
-            streamFn().buffered(16).use { stream ->
-                // SY -->
-                (if (extraPage != null) streamFn2?.invoke()?.buffered(16) else null).use { stream2 ->
-                    if (viewer.config.dualPageSplit) {
-                        process(item.first, stream)
-                    } else {
-                        mergePages(stream, stream2)
-                    }.use { itemStream ->
-                        // SY <--
-                        val bais = ByteArrayInputStream(itemStream.readBytes())
-                        val isAnimated = ImageUtil.isAnimatedAndSupported(bais)
-                        bais.reset()
-                        val background = if (!isAnimated && viewer.config.automaticBackground) {
-                            ImageUtil.chooseBackground(context, bais)
+        // SY -->
+        try {
```

Comment:
Whats the reason for this? Is there a common crash here or something?

Reply:
The app will hard crash here if the archive password is wrong.

Reply:
Do you think you could test this before the reader is fully open, so it can automatically close the reader if the password is invalid?

Reply:
It could be tested in the ZipChapterLoader, so it would work between chapters

Reply:
I changed it